Feature off days management

lomamech opened the pull request:
feature(Offdays Management)  …
- Implement API for OffDaymanagement
- (test) Implement test Integration for Offday API
- Implementation of client side for Offdays Unit
- Implement test End to End for Offdays Management

reference #2292

lomamech committed:
test(Offday Management)
- Implement test End to End for Offdays Management

lomamech committed:
test(Offday Management)
- Implemen test E2E for Offday Management

jniles commented on client/src/modules/offdays/modals/offday.modal.html:
```diff
@@ -0,0 +1,49 @@
+    </div>
+
+    <div class="modal-body" ng-class="{ 'has-error' : OffdayForm.$submitted && OffdayForm.percent_pay.$invalid }">
+      <label class="control-label" translate>FORM.LABELS.PERCENTAGE</label>
```
It might be clearer if you include the percent sign here.  Like this:
```html
<label class="control-label">
  <span translate>FORM.LABELS.PERCENTAGE</span> (%)
</label>
```

jniles commented on client/src/modules/offdays/modals/offday.modal.js:
```diff
@@ -0,0 +1,58 @@
+  '$state', 'OffdayService', 'ModalService', 'NotifyService', 'appcache', 'moment',
+];
+
+function OffdayModalController($state, Offdays, ModalService, Notify, AppCache, moment) {
```
Do you use `ModalService` here?

jniles commented on client/src/modules/offdays/modals/offday.modal.js:
```diff
@@ -0,0 +1,58 @@
+      .catch(Notify.handleError);
+  }
+
+  function closeModal() {
```
Can you just use `ui-sref="offdays"` here?

jniles commented on client/src/modules/offdays/offdays.html:
```diff
@@ -0,0 +1,44 @@
+
+    <div class="toolbar">
+      <div class="toolbar-item">
+        <button class="btn btn-default text-capitalize" id="create" ng-click="OffdayCtrl.createOffday()" data-method="create">
```
Do you need the `id="create"` here?

jniles commented on client/src/modules/offdays/offdays.js:
```diff
@@ -0,0 +1,104 @@
+    [
+    
+      { field : 'label', displayName : 'FORM.LABELS.DESIGNATION', headerCellFilter : 'translate' },
+      { field : 'date', displayName : 'FORM.LABELS.DATE', cellFilter : 'date:"mediumDate"', headerCellFilter : 'translate' },
```
Instead of `date:"mediumDate"` here, could you use `bhConstants`?  That way all the dates render the same way across browsers and operating systems.

jniles commented on client/src/modules/offdays/offdays.js:
```diff
@@ -0,0 +1,104 @@
+  function loadOffdays() {
+    vm.loading = true;
+
+    Offdays.read(null, { detailed : 1 })
```
What does `detailed: 1` do?  It doesn't look like it is used.

jniles commented on client/src/modules/offdays/offdays.service.js:
```diff
@@ -0,0 +1,17 @@
+
+/**
+ * @class OffdayService
+ * @extends PrototypeApiService
```
:+1:

jniles commented on test/integration/offdays.js:
```diff
@@ -0,0 +1,83 @@
+ *
+ * This test suite implements full CRUD on the /offdays  HTTP API endpoint.
+ */
+describe('(/offdays) The /offdays  API endpoint', function () {
```
Could you make these integrations tests respect ESLint?

jniles reviewed (commented):
A lot of these are very similar to the comments on #2326.  Good work porting this code!

lomamech committed:
refactor(Offday Management)
- Set sign of percent on form
- Remove unused property, sanitaze code
- Use ui-sref for redirection
- Sanitaze integration test with eslint

lomamech committed:
Remove unused function

lomamech committed:
perf(OffDay Management)
- Use concat in integration test of OffDay instead signe plus

lomamech committed:
Fix and Resolve Conflict

lomamech committed:
Resolve Conflict with Branch Master
